rudder-lang/src/logger.rs

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+use log::LevelFilter;
+use core::str::FromStr;
+use std::io::Write;
+use std::panic;
+
+use std::path::PathBuf;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+/// Adds verbose levels: off, error, (warn,) info, debug, (trace). For example info includes error, debug includes info and error
+/// The level is set through program arguments. Default is Warn
+/// run the program with `-l info` (eq. `--logger info`) optional argument. Case-insensitive
+/// There is also an optional json formatter that outputs plain json format. run the program with `-j` or `--json` optional argument.
+pub fn set(log_level_str: &str, is_json: bool) {
+    let log_level = match LevelFilter::from_str(log_level_str) {
+        Ok(level) => level,
+        Err(_) => LevelFilter::Warn 
+    };
+
+    // Content called when panic! is encountered to close logger brackets and print error
+    set_panic_hook(is_json);
+
+    if is_json {
+        print_json_output_opener();
+        // prevents any output stylization from the colored crate
+        colored::control::set_override(false);
+    }
+    
+    let mut builder = env_logger::Builder::new();
+    if is_json {
+        // Note: record .file() and line() allow to get the origin of the print
+        builder.format(move |buf, record| {
+            writeln!(buf, r#"    {{
+      "status": "{}",
+      "message": "{}"
+    }},"#,
+            record.level().to_string().to_ascii_lowercase(),
+            record.args().to_string()
+        )});
+    }
+    builder.filter(None, log_level)
+    .format_timestamp(None)
```

Comment:
missing rustfmt for this

Reply:
I'll apply rustfmt on the entire project in a separated PR to have a standard and cleaned format